agenta-backend/agenta_backend/services/db_manager.py

```diff
@@ -360,6 +380,7 @@ async def remove_image(image: Image, **kwargs: dict):
         query.eq(ImageDB.tags, image.tags)
         & query.eq(ImageDB.docker_id, image.docker_id)
         & query.eq(ImageDB.user_id, user.id)
+        & query.eq(ImageDB.id, ObjectId(image.id))
```

Comment:
This constitutes the first cause of the bug. 

If this condition is omitted, the process of deleting an app results in the deletion of the first image that resolves the query expression. Consequently, if a user has 2 apps and chooses to delete one, the other app is also affected, leading to a backend error indicating a referenced document not found.